src/esp/gfx/CMakeLists.txt

```diff
@@ -76,6 +78,19 @@ target_include_directories(gfx
     "${DEPS_DIR}/glad/include"
 )
 
+if(BUILD_WITH_CUDA)
+  find_package(CUDA 8.0)
+  if(CUDA_FOUND)
```

Comment:
I'm a fan of things either reliably working or reliably not working, so instead of silently building without CUDA when it's not found, I would

```cmake
if(NOT CUDA_FOUND)
    message(FATAL_ERROR "CUDA was not found. Either install it or disable BUILD_WITH_CUDA.")
endif()
```

... because otherwise the message about CUDA not being found gets lost somewhere wayy up in the log output and users wouldn't know that until they try using those APIs.

Same goes for the `ESP_BUILD_WITH_CUDA` done in `core/CMakeLists.txt` -- actually, I'd move this whole thing one level up so the finding and checking is done just on a single place.